src/Artsy/Router/buildServerApp.tsx

```diff
@@ -175,24 +175,35 @@ export function buildServerApp(config: ServerRouterConfig): Promise<Resolve> {
               scripts.push(
                 bundleScriptTags
                   .split("\n")
-                  .filter(
-                    script =>
-                      !(
-                        /**
-                         * Since these files are already embedded in our main
-                         * layout file, omit them from the scripts array.
-                         *
-                         * TODO: Don't include these files in the main layout
-                         * and instead relay on dynamically including them here.
-                         * Will require some shuffling in the jade template,
-                         * however.
-                         */
-                        (
-                          script.includes("/assets/runtime-manifest.") ||
-                          script.includes("/assets/common.")
-                        )
+                  .map(script => {
+                    // In production, prefix injected script src with CDN endpoint
+                    if (getENV("CDN_URL")) {
+                      const scriptTagWithCDN = script.replace(
+                        /src="\/assets/g,
+                        `src="${getENV("CDN_URL")}/assets`
```

Comment:
Is there a way this can use exactly the same code as is used elsewhere to render the "main" asset hostname? Otherwise we risk generating the URLs slightly differently, or allowing the 2 locations to get out-of-sync over time

Reply:
Unfortunately not, since we primarly handle our script injection in force, in a jade [asset helper middleware](https://github.com/artsy/force/blob/master/src/lib/middleware/assetMiddleware.ts#L23).

Coming from reaction, we pass through different kinds of javascript, not just script tags pointing at urls. This can be refactored so that a distinction can be made, however, but that would be follow-up. 

In any case, the logic is the same in both cases: `if CDN_URL, prepend`